packages/react-devtools-shared/src/backend/fiber/renderer.js

```diff
@@ -1269,51 +1281,66 @@ export function attach(
     if (__DEBUG__) {
       if (didGenerateID) {
         debug(
-          'getOrGenerateFiberID()',
+          'getOrGenerateFiberInstance()',
           fiber,
-          fiber.return,
+          fiberInstance.parent,
           'Generated a new UID',
         );
       }
     }
 
-    return fiberInstance.id;
+    return fiberInstance;
+  }
+
+  // Returns a FiberInstance if one has already been generated for the Fiber or throws.
+  function getFiberInstanceThrows(fiber: Fiber): FiberInstance {
+    const fiberInstance = getFiberInstanceUnsafe(fiber);
+    if (fiberInstance !== null) {
+      return fiberInstance;
+    }
+    throw Error(
+      `Could not find ID for Fiber "${getDisplayNameForFiber(fiber) || ''}"`,
+    );
   }
 
-  // Returns an ID if one has already been generated for the Fiber or throws.
   function getFiberIDThrows(fiber: Fiber): number {
-    const maybeID = getFiberIDUnsafe(fiber);
-    if (maybeID !== null) {
-      return maybeID;
+    const fiberInstance = getFiberInstanceUnsafe(fiber);
+    if (fiberInstance !== null) {
+      return fiberInstance.id;
     }
     throw Error(
       `Could not find ID for Fiber "${getDisplayNameForFiber(fiber) || ''}"`,
     );
   }
 
-  // Returns an ID if one has already been generated for the Fiber or null if one has not been generated.
+  // Returns a FiberInstance if one has already been generated for the Fiber or null if one has not been generated.
   // Use this method while e.g. logging to avoid over-retaining Fibers.
-  function getFiberIDUnsafe(fiber: Fiber): number | null {
+  function getFiberInstanceUnsafe(fiber: Fiber): FiberInstance | null {
     const fiberInstance = fiberToFiberInstanceMap.get(fiber);
     if (fiberInstance !== undefined) {
-      return fiberInstance.id;
+      return fiberInstance;
     } else {
       const {alternate} = fiber;
       if (alternate !== null) {
         const alternateInstance = fiberToFiberInstanceMap.get(alternate);
         if (alternateInstance !== undefined) {
-          return alternateInstance.id;
+          return alternateInstance;
         }
       }
     }
     return null;
   }
 
+  function getFiberIDUnsafe(fiber: Fiber): number | null {
+    const fiberInstance = getFiberInstanceUnsafe(fiber);
+    return fiberInstance === null ? null : fiberInstance.id;
+  }
+
   // Removes a Fiber (and its alternate) from the Maps used to track their id.
   // This method should always be called when a Fiber is unmounting.
   function untrackFiberID(fiber: Fiber) {
     if (__DEBUG__) {
-      debug('untrackFiberID()', fiber, fiber.return, 'schedule after delay');
+      debug('untrackFiberID()', fiber, null, 'schedule after delay');
```

Comment:
Why `fiber.return` is removed here? Because it should ideally point to `DevToolsInstance`, which can potentially be `VirtualInstane`?

Reply:
It’s a bit unclear what this is meant to help catch. This should be null already I believe. So is it meant to be like the previous parent?

Reply:
> It’s a bit unclear what this is meant to help catch. This should be null already I believe. So is it meant to be like the previous parent?

Its kind of useless, I believe we just print it because of the spec of `debug` function.  Could there be non-null reference to return Fiber? Is it guaranteed that `return` is going to be nullified for every Fiber in an unmounted branch by reconciler?